src/Plugin/Block/PageHeaderBlock.php

```diff
@@ -152,6 +155,13 @@ public function __construct(array $configuration, $plugin_id, $plugin_definition
           }
         }
       }
+      if (isset($defaults['view_id']) && isset($defaults['display_id'])) {
+        $view = Views::getView($defaults['view_id']);
+        if ($view) {
+          $view->setDisplay($defaults['display_id']);
+          $this->entity = $view;
```

Comment:
Not of type Drupal\Core\Entity\EntityInterface. 
The pageHeaderEvent isn't doing type checking (maybe it should) so perhaps we need to send $view separately?